1.4.1: pytest id failing and warnings

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

After add `flaky` 3.7.0 to BuildRequires I have now 3 units failing
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-subprocess-1.4.1-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-subprocess-1.4.1-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-subprocess-1.4.1, configfile: setup.cfg
plugins: subprocess-1.4.1, asyncio-0.19.0, flaky-3.7.0, anyio-3.6.1
asyncio: mode=strict
collected 123 items

tests/test_asyncio.py ..........ssss.............                                                                                                                    [ 21%]
tests/test_examples.py ..                                                                                                                                            [ 23%]
tests/test_subprocess.py ....................................EE...................E.......................                                                           [ 89%]
tests/test_typing.py .                                                                                                                                               [ 90%]
tests/test_utils.py ............                                                                                                                                     [100%]

================================================================================== ERRORS ==================================================================================
_______________________________________________________________ ERROR at setup of test_multiple_wait[False] ________________________________________________________________

self = <flaky.flaky_pytest_plugin.FlakyPlugin object at 0x7ff2b9f1f610>, item = <Function test_multiple_wait[False]>

    def pytest_runtest_setup(self, item):
        """
        Pytest hook to modify the test before it's run.

        :param item:
            The test item.
        """
        if not self._has_flaky_attributes(item):
            if hasattr(item, 'iter_markers'):
                for marker in item.iter_markers(name='flaky'):
>                   self._make_test_flaky(item, *marker.args, **marker.kwargs)
E                   TypeError: _make_test_flaky() got an unexpected keyword argument 'reruns'

/usr/lib/python3.8/site-packages/flaky/flaky_pytest_plugin.py:239: TypeError
________________________________________________________________ ERROR at setup of test_multiple_wait[True] ________________________________________________________________

self = <flaky.flaky_pytest_plugin.FlakyPlugin object at 0x7ff2b9f1f610>, item = <Function test_multiple_wait[True]>

    def pytest_runtest_setup(self, item):
        """
        Pytest hook to modify the test before it's run.

        :param item:
            The test item.
        """
        if not self._has_flaky_attributes(item):
            if hasattr(item, 'iter_markers'):
                for marker in item.iter_markers(name='flaky'):
>                   self._make_test_flaky(item, *marker.args, **marker.kwargs)
E                   TypeError: _make_test_flaky() got an unexpected keyword argument 'reruns'

/usr/lib/python3.8/site-packages/flaky/flaky_pytest_plugin.py:239: TypeError
___________________________________________________________ ERROR at setup of test_raise_exception_check_output ____________________________________________________________

self = <flaky.flaky_pytest_plugin.FlakyPlugin object at 0x7ff2b9f1f610>, item = <Function test_raise_exception_check_output>

    def pytest_runtest_setup(self, item):
        """
        Pytest hook to modify the test before it's run.

        :param item:
            The test item.
        """
        if not self._has_flaky_attributes(item):
            if hasattr(item, 'iter_markers'):
                for marker in item.iter_markers(name='flaky'):
>                   self._make_test_flaky(item, *marker.args, **marker.kwargs)
E                   TypeError: _make_test_flaky() got an unexpected keyword argument 'reruns'

/usr/lib/python3.8/site-packages/flaky/flaky_pytest_plugin.py:239: TypeError
============================================================================= warnings summary =============================================================================
tests/test_examples.py::test_documentation[docs/index.rst]
tests/test_examples.py::test_documentation[README.rst]
  /home/tkloczko/rpmbuild/BUILD/pytest-subprocess-1.4.1/tests/test_examples.py:17: PendingDeprecationWarning: nodes.Node.traverse() is obsoleted by Node.findall().
    code_blocks = publish_doctree(content).traverse(condition=is_code_block)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [4] tests/test_asyncio.py:110: condition: sys.platform!="win32"
ERROR tests/test_subprocess.py::test_multiple_wait[False] - TypeError: _make_test_flaky() got an unexpected keyword argument 'reruns'
ERROR tests/test_subprocess.py::test_multiple_wait[True] - TypeError: _make_test_flaky() got an unexpected keyword argument 'reruns'
ERROR tests/test_subprocess.py::test_raise_exception_check_output - TypeError: _make_test_flaky() got an unexpected keyword argument 'reruns'
=========================================================== 116 passed, 4 skipped, 2 warnings, 3 errors in 6.46s ===========================================================
```

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -1,6 +1,24 @@
 History
 =======
 
+unreleased (2022-10-02)  
+-----------------------
+
+Features  
+~~~~~~~~
+* `#87 <https://github.com/aklajnert/pytest-subprocess/pull/87>`_: Add support for Python 3.11.
+* `#80 <https://github.com/aklajnert/pytest-subprocess/pull/80>`_, `#86 <https://github.com/aklajnert/pytest-subprocess/pull/86>`_: The `register()` method returns an auxiliary object that will contain all matching `FakePopen` instances.
+
+Bug fixes  
+~~~~~~~~~
+* `#93 <https://github.com/aklajnert/pytest-subprocess/pull/93>`_: Raise callback exceptions on `communicate()` calls.
+
+Other changes  
+~~~~~~~~~~~~~
+* `#97 <https://github.com/aklajnert/pytest-subprocess/pull/97>`_: Fixed warnings in tests, treat warnings as errors.
+* `#91 <https://github.com/aklajnert/pytest-subprocess/pull/91>`_: Use `sys.executable` instead just `"python"` in tests while invoking python subprocess.
+* `#90 <https://github.com/aklajnert/pytest-subprocess/pull/90>`_: Fix documentation build, add CI check for it.
+
 1.4.1 (2022-02-09)  
 ------------------
 

--- a/changelog.d/other.533cbb9f.entry.yaml
+++ b/changelog.d/other.533cbb9f.entry.yaml
@@ -1,0 +1,5 @@
+message: Fixed warnings in tests, treat warnings as errors.
+pr_ids:
+- '97'
+timestamp: 1664691594
+type: other

--- a/pytest.ini
+++ b/pytest.ini
@@ -3,3 +3,4 @@ junit_family=legacy
 ;asyncio_mode=auto
 filterwarnings =
     error
+    ignore::pytest.PytestUnraisableExceptionWarning

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,6 +1,5 @@
 [pytest]
 junit_family=legacy
-;asyncio_mode=auto
 filterwarnings =
     error
     ignore::pytest.PytestUnraisableExceptionWarning

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,5 +1,5 @@
 [pytest]
 junit_family=legacy
-asyncio_mode=auto
+;asyncio_mode=auto
 filterwarnings =
     error

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,0 +1,5 @@
+[pytest]
+junit_family=legacy
+asyncio_mode=auto
+filterwarnings =
+    error

--- a/setup.cfg
+++ b/setup.cfg
@@ -11,6 +11,3 @@ ignore_missing_imports = True
 
 [mypy-setup]
 ignore_errors = True
-
-[tool:pytest]
-junit_family=legacy

--- a/tests/test_examples.py
+++ b/tests/test_examples.py
@@ -14,7 +14,7 @@ def get_code_blocks(file_path):
     with file_path.open() as file_handle:
         content = file_handle.read()
 
-    code_blocks = publish_doctree(content).traverse(condition=is_code_block)
+    code_blocks = publish_doctree(content).findall(condition=is_code_block)
     return [block.astext() for block in code_blocks]
 
 

